modules/karma.py

```diff
@@ -70,28 +70,29 @@ def sendmsg(sender, recip, text):
         except:
             return "Ma di che parli?"
     elif (text.endswith('++') and len(text.split(' ')) == 1):
-        nick = text[:-2]
-        if karma == config['nickname']:
-            karma[nick] = readval(nick) + 1
-            result = "Grazie per la tua stima"
-        elif nick == sender:
-            result = "Un po' autoreferenziale, non credi?"
-        else:
-            karma[nick] = readval(nick) + 1
-            result = "Prendo nota. %s: %d" % (nick, karma[nick])
-        save()
-        return result
+        return vote(text[:-2])
     elif (text.endswith('--') and len(text.split(' ')) == 1):
-        nick = text[:-2]
-        if nick == config['nickname']:
-            result = "Nah, non credo di volerlo fare"
-        else:
-            karma[nick] = readval(nick) - 1
-            result = "Prendo nota. %s: %d" % (nick, karma[nick])
-        save()
-        return result
+        return vote(text[:-2], -1)
     return None
 
+def vote(nick, delta=1):
```

Comment:
You find the interface confusing? It's not needed, so go ahead and make it a parameter.